catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/service/SearchService.java

```diff
@@ -269,7 +261,20 @@ private Query createQuery(Filter filter, Long startIndexLng, Long countLng, Stri
         String sortField = Result.TEMPORAL;
         String sortOrder = DEFAULT_SORT_ORDER;
         Long startIndex = startIndexLng == null ? Long.valueOf(DEFAULT_START_INDEX) : startIndexLng;
-        Long count = countLng == null ? Long.valueOf(DEFAULT_COUNT) : countLng;
+        int count;
+        if (countLng == null) {
+            count = DEFAULT_COUNT;
+        } else if (countLng > searchController.getMaxQueryResults()) {
+            count = searchController.getMaxQueryResults();
+            LOGGER.info(
```

Comment:
❓ Since we could we end up flooding the log file if a client keeps using a count that's too high, are we sure we want to log this at info?